### DISCUSSION TITLE
Nodes running out of space

### DISCUSSION THREAD

**Discussion by carstenblt** [mysticaltech/terraform-hcloud-kube-hetzner#1052] [Q&A] [answered]
### Description

My nodes just ran out of space because containerd kept all old images. Running

```
crictl rmi --prune
```

solved it. I am not sure yet about the best way to run this regularly but it might be worth adding?

**Comment by Silvest89** [comment, marked as answer]
Kubernetes automatically performs garbage collection
https://kubernetes.io/docs/concepts/architecture/garbage-collection/#containers-images

You can tune using the following kubelet args
* image-gc-high-threshold=80
* image-gc-low-threshold=75

**Comment by carstenblt** [comment]
Oh, I see, thank you!

So
```
k3s_exec_server_args = "--kubelet-arg=image-gc-high-threshold=50 --kubelet-arg=image-gc-low-threshold=40"
```

should do the trick?

**Comment by Silvest89** [comment]
Don't forget that you need to do the same for agent_args

And using those args wont work for existing nodes, and they won't persist across reboots.
I have opened a PR that allows to set kubelet-arg in the config.yaml instead

https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/1044

**Comment by maggie44** [comment]
Aren’t those args the default? https://github.com/k3s-io/k3s/issues/813
